### PR TITLE
feat: add repository search to the main page

### LINKED ISSUE #91: Feature: Add Repository Search
### Type of feature

🍕 Feature

### Current behavior

Repositories are displayed based on the Links at the top of the page

image

These links filter the repositories based on pre-defined filters. Currently, if a repository is outside of these filters it will never be displayed on the page making it unavailable for actions such as upvoting. 

### Suggested solution

Add the ability to search for a repository so that queries outside of the filters can be found. 

This will require an update to the UI to add a search box to allow the user to type in a repository to search for. A couple of things to consider for this functionality:

- Will the search bar always be visible or only when the user wants to search?
- It would be nice if the search returns repositories while the user is typing. This should implement the concept of debouncing to limit the number of requests to the database. This [article](https://www.freecodecamp.org/news/javascript-debounce-example/) gives a good description of this concept and how to implement it. 
 

### Additional context

No additial no context

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by hokagedemehin**
## What type of PR is this? (check all applicable)

- [x] 🍕 Feature

## Description

This PR adds the search component to the main page of the web application. Currently, it does not have any function linked to it, so it does not filter out the page based on what you type. I will need @bdougie help with that

## Related Tickets & Documents
Fixes #91 

## Mobile & Desktop Screenshots/Recordings

Screen Shot 2022-02-25 at 3 23 13 PM

## Added tests?
- [x] 🙅 no, because they aren't needed

## Added to documentation?
- [x] 🙅 no documentation needed

## [optional] What gif best describes this PR or how it makes you feel?
denzel

### CODE REVIEW

**Comment by bdougie** [comment]
added a screenshot ;)

**Review by bdougie — CHANGES_REQUESTED** [review]
Screen Shot 2022-02-25 at 3 26 50 PM

When logged it looks like the search box overlaps. Be sure to grab the latest, because you can now log in on localhost.

File: src/components/PrimaryNav.jsx [hunk 1]
```diff
@@ -8,30 +8,52 @@ const PrimaryNav = () => {
...
         sauced"
             src={logo}
           />
         </a>
       </div>
+      <div className="flex justify-center items-center space-x-2 sm:space-x-3">
+        <input
+          className=" ring-1 rounded-lg shadow-md w-[120px] h-full p-2 text-[9px] sm:text-xs sm:w-40 "
+          type="search"
+          placeholder="search open sauced 🔍"
```
Comment (bdougie) [inline src/components/PrimaryNav.jsx:26]: We can lose the 🔎

File: src/components/PrimaryNav.jsx [hunk 2]
```diff
@@ -8,30 +8,52 @@ const PrimaryNav = () => {
...
+              }}
+            >
+              Login
+            </div>
+          </div>
+        )}
+      </div>
 
-      {!user && (
+      {/* {!user && (
```
Comment (bdougie) [inline src/components/PrimaryNav.jsx:45]: what is up with the comments here?

Comment (hokagedemehin) [reply src/components/PrimaryNav.jsx:45]: for some weird reason, the logic appeared twice in my files. Maybe I will just remove it

File: src/components/PrimaryNav.jsx (hunk 1 above)
Comment (hokagedemehin) [reply src/components/PrimaryNav.jsx]: I have removed the search icon

**Comment by hokagedemehin** [comment]
> Screen Shot 2022-02-25 at 3 26 50 PM
> 
> When logged it looks like the search box overlaps. Be sure to grab the latest, because you can now log in on localhost.

I have also corrected the overlapping issue, it is properly spaced now

File: src/components/PrimaryNav.jsx (hunk 2 above)
Comment (bdougie) [reply src/components/PrimaryNav.jsx]: Yes, please. If it is needed we can add it back but currently it is dead code.

Comment (hokagedemehin) [reply src/components/PrimaryNav.jsx]: I have also removed the code you highlighted here too

**Comment by bdougie** [comment]
Cherry-picking this thing over here #155. Thanks for the work so far, I will make sure your commits are referenced on my PR.

**Pull request closed without merging**